plugins/kinetic-dp/fu-plugin-kinetic-dp.c

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright (C) 2017 Mario Limonciello <mario.limonciello@dell.com>
+ * Copyright (C) 2017 Peichen Huang <peichenhuang@tw.kinetic.com>
+ * Copyright (C) 2017 Richard Hughes <richard@hughsie.com>
+ * Copyright (C) 2021 Jeffrey Lin <jlin@kinet-ic.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include <fwupdplugin.h>
+
+#include "fu-kinetic-dp-common.h"
+#include "fu-kinetic-dp-device.h"
+#include "fu-kinetic-dp-firmware.h"
+
+#define FU_KINETIC_DP_DRM_REPLUG_DELAY	5	/* sec */
+
+struct FuPluginData {
+	GPtrArray	*devices;
+	guint		drm_changed_id;
+};
+
```

Comment:
so the previous codebase this was plugin was partially based upon had a safety check for `amdgpu`.  Does that apply here?  Is it possible that one of these DP converter chips is connected to an `amdgpu` with an older kernel?  Or is it only in fixed hardware designs (IE soldered down to the board and running in stuff like ChromeOS)?  

@hughsie since we now _potentially_ have 3 plugins potentially accessing DP aux, I think it's worth moving the safety check "into the daemon" for that somewhere. Thoughts?

Reply:
Currently we can't confirm  what kind of GPU would be used on our customer's product.
But if this is a known common issue related to kernel version, I think we should also include this safety check as well.

Reply:
OK, we need to pull this into the daemon then separately from this PR and modify to make sure all plugins using DP aux do the safety check from the daemon.